components/Calendar/CalendarDayView.js

```diff
@@ -0,0 +1,111 @@
+import Head from 'next/head';
+import { useState, useEffect } from 'react';
+import { styled } from '@mui/material/styles';
+import Paper from '@mui/material/Paper';
+import { ViewState } from '@devexpress/dx-react-scheduler';
+import { Scheduler, DayView, Toolbar, DateNavigator, Appointments, TodayButton, AppointmentTooltip } from '@devexpress/dx-react-scheduler-material-ui';
+
+import TooltipContent from '../Calendar/subcomponents/TooltipContent.js';
+import styles from '../../utils/styles/CalendarStyles/Calendar.module.css';
+import { TimeTableCell, DayScaleCell } from './Calendar.js';
+import CreateClassModal from '../CreateClassModal/CreateClassModal.js';
+
+import { getCoursesByMentorId, getCoursesByMenteeId, getUserInfo } from '../../utils/api/apiCalls.js';
+
+
+export default function Calendar({ userInfo, formattedAllCoursesData }) {
+  const [appointmentData, setAppointmentData] = useState(formattedAllCoursesData);
+  const [userType, setUserType] = useState(userInfo.account_type);
+  const [currUserId, setCurrUserId] = useState(userInfo.id);
+
+  const getCoursesData = () => {
+    if (userType === 'Mentor') {
+      getCoursesByMentorId(currUserId)
+        .then(res => {
+          let apptDataResult = res.map((course) => {
+            return {
+              title: course.name,
+              startDate: new Date(course.start_date),
+              endDate: new Date(course.end_date),
+              zoomLink: course.meeting_url,
+              capacity: course.capacity,
+              description: course.description,
+              id: course.id,
+              subject: course.subject,
+              photo: course.photo,
+              mentor: course.mentor,
+              mentees: course.mentees,
+            }
+          })
+          // console.log(apptDataResult);
+          setAppointmentData(apptDataResult);
+        })
+        .catch(err => {
+          console.log(err);
+        })
+    } else {
+      getCoursesByMenteeId(currUserId)
```

Comment:
Could be a way to make this more dynamic. Could do something like 
`let courseFetch = (userType === 'Mentor') ? getCoursesByMentorId : getCoursesByMenteeId;
courseFetch(currUserId).then(...).catch(...)`
This would save you from doing that huge map twice and only have to do once

Reply:
Noted. Will refactor rn!